src/Parser.php

```diff
@@ -435,15 +456,17 @@ protected function isHostLabel($label): bool
      */
     protected function filterPort($port)
     {
-        if (in_array($port, [null, false, ''], true)) {
+        if (null === $port || false === $port || '' === $port) {
             return null;
         }
 
-        if (!$this->isPort($port)) {
+        $result = filter_var($port, FILTER_VALIDATE_INT, ['options' => ['min_range' => 1, 'max_range' => 65535]]);
```

Comment:
referring to issue #15 we can simplify this code using a simple regex to check that the port contains only digit. I believe this means that port `0` is accepted too but negative port are still forbidden